src/networking.c

```diff
@@ -1283,6 +1292,43 @@ void addReplySubcommandSyntaxError(client *c) {
     sdsfree(cmd);
 }
 
+inline int isDeferredReplyEnabled(client *c) {
+    return c->deferred_reply_bytes != ULLONG_MAX;
+}
+
+void initDeferredReplyBuffer(client *c) {
+    if (moduleNotifyKeyspaceSubscribersCnt() == 0) return;
+    if (c->deferred_reply == NULL) c->deferred_reply = listCreate();
+    if (!isDeferredReplyEnabled(c)) c->deferred_reply_bytes = 0;
+}
+
+static void resetDeferredReplyBuffer(client *c) {
+    listEmpty(c->deferred_reply);
+    c->deferred_reply_bytes = ULLONG_MAX;
+}
```

Comment:
Making this a no-op if there are no modules subscribing to keyspace notifications is a smart idea. Almost too smart, so contributors will not understand why we do this. :smile:

Let's add some comment to document these functions. Especially `initDeferredReplyBuffer`, why some commands should use it and why module keyspace subscribers have anything to do with it.

Something like this? Correct it if it's wrong...

```C
/* Commands that produce replies before signaling keyspace notifications need
 * to use a deferred reply buffer to ensure read-after-write consistency.
 * This is because modules can be subscribing to keyspace notifications and
 * block the client from the keyspace notification callback. */
void initDeferredReplyBuffer(client *c) {
```